tests/js/common/shell/shell-collection-api.js

```diff
@@ -0,0 +1,1691 @@
+          validateProperties({replicationFactor: 3, minReplicationFactor: 2, writeConcern: 2}, collname, 2);
+        } else {
+          validateProperties({replicationFactor: 3, writeConcern: 1}, collname, 2);
+        }
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testGloballyUniqueIdReal: function () {
+      const globallyUniqueId = "h843EE8622105/182";
+      const res = tryCreate({name: collname, globallyUniqueId});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        validateProperties({}, collname, 2);
+        validatePropertiesAreNotEqual({globallyUniqueId}, collname);
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testGloballyUniqueIdOtherString: function () {
+      const res = tryCreate({name: collname, globallyUniqueId: "test"});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        validateProperties({}, collname, 2);
+        validatePropertiesAreNotEqual({globallyUniqueId: "test"}, collname);
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testGloballyUniqueIdNumeric: function () {
+      const res = tryCreate({name: collname, globallyUniqueId: 123456});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        validateProperties({}, collname, 2);
+        validatePropertiesAreNotEqual({globallyUniqueId: 123456}, collname);
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testGloballyUniqueIdCreateTwice: function () {
+      const globallyUniqueId = "notSoUnique";
+      const res = tryCreate({name: collname, globallyUniqueId});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        // Make sure first collection does not have this id
+        validateProperties({}, collname, 2);
+        validatePropertiesAreNotEqual({globallyUniqueId}, collname);
+        const otherCollName = `${collname}_2`;
+        const res2 = tryCreate({name: otherCollName, globallyUniqueId});
+        try {
+          // We always generate a new ID
+          assertTrue(res2.result, `Result: ${JSON.stringify(res2)}`);
+          validateProperties({}, otherCollName, 2);
+          validatePropertiesAreNotEqual({globallyUniqueId}, otherCollName);
+        } finally {
+          db._drop(otherCollName);
+        }
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testIsSmartStandAlone: function () {
+      const res = tryCreate({name: collname, isSmart: true});
+      try {
+        if (!isEnterprise) {
+          assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+          // isSmart is just ignored, we keep default: false
+          validateProperties({}, collname, 2);
+          validateDeprecationLogEntryWritten();
+          if (!isCluster) {
+            // SingleSever does not expose is Smart at all
+            validatePropertiesDoNotExist(collname, ["isSmart"]);
```

Comment:
Yes we do,
and if you create a real SmartVertexCollection (see: testCreateSmartGraph) the value isSmart: true is exposed in Enterprise mode on Single and Cluster.
It seems that on SingleServer the value can only be "true" or not exist.
On Cluster it can be only be "false" or "true".

At least that is what Is exposed via the properties API.